contributed/hackurdc2_utils/cosmodc2realizer_helper.py

```diff
@@ -0,0 +1,149 @@
+    for component in ['disk', 'bulge']:
+        galaxies['ra_%s' %component] = galaxies['ra'].values
+        galaxies['dec_%s' %component] = galaxies['dec'].values
+        galaxies['size_circular_%s' %component] = (galaxies['size_minor_%s_true' %component].values*galaxies['size_%s_true' %component].values)**0.5
+        galaxies['e_%s' %component] = galaxies['ellipticity_%s_true' %component]
+
+    galaxies_cols_to_keep = ['galaxy_id', 'ra', 'dec'] # 'agn', 'sprinkled', 'star']
+    galaxies_cols_to_keep += ['ra_bulge', 'dec_bulge', 'size_circular_bulge', 'e_bulge', 'phi_bulge']
+    galaxies_cols_to_keep += ['ra_disk', 'dec_disk', 'size_circular_disk', 'e_disk', 'phi_disk'] 
+    galaxies_cols_to_keep += [prop + '_' + bp for prop, bp in product(['flux_bulge', 'flux_disk'], filters)]
+    galaxies = galaxies[galaxies_cols_to_keep]
+    if save_to_disk is not None:
+        galaxies.to_csv(save_to_disk, index=False)
+    return galaxies
+
+def _format_truth_catalog(point_neighbors, save_to_disk=None):
+    # Point-source neighbors
+    point_neighbors[['ra', 'dec']] = units.deg_to_arcsec(point_neighbors[['ra', 'dec']])
+    for bp in 'ugrizy':
+        point_neighbors['flux_%s' %bp] = units.mag_to_flux(point_neighbors[bp].values, to_unit='nMgy')
+    if save_to_disk is not None:
+        point_neighbors.to_csv(save_to_disk, index=False)
+    return point_neighbors
+
+def get_neighbors(candidate_df, reference_ra, reference_dec, radius):
+    from scipy import spatial
+    positions = np.c_[candidate_df[['ra', 'dec']].values]
+    tree = spatial.cKDTree(positions)
+    target_objects_idx = tree.query_ball_point(x=[reference_ra, reference_dec], r=radius, p=2)
+    target_objects = candidate_df.iloc[target_objects_idx].copy()
+    return target_objects, target_objects_idx
+
+def separate_bulge_disk(extragal_df):
+    # Rename for convenience
+    df = extragal_df
+    # Separate df into bulge-related and disk-related
+    bulge_df = df.filter(like='bulge', axis=1).copy()
+    disk_df = df.filter(like='disk', axis=1).copy()
+    # Make column schema the same across bulge and disk DataFrames (not sure if necessary)
+    bulge_df.columns = [col.strip().replace('_bulge', '') for col in bulge_df.columns]
+    disk_df.columns = [col.strip().replace('_disk', '') for col in disk_df.columns]
+    return bulge_df, disk_df, df
+
+def point_to_mog(point_df):
+    point_df['gauss_sigma'] = 0.0
+    point_df['e'] = 0.0
+    point_df['phi'] = 0.0
+    # Column sanity check
+    output_cols = ['ra', 'dec', 'e', 'phi', 'gauss_sigma',]
+    output_cols += ['flux_%s' %bp for bp in 'ugrizy']
+    mog_df = point_df[output_cols]
+    return mog_df
+
+def sersic_to_mog(sersic_df, bulge_or_disk):
+    from scipy.special import gammaincinv
+    if bulge_or_disk=='bulge':
+        # Mixture of gaussian parameters for de Vaucouleurs profile from HL13
+        weights = [0.00139, 0.00941, 0.04441, 0.16162, 0.48121, 1.20357, 2.54182, 4.46441, 6.22821, 6.15393]
```

Comment:
Given that this util function will be separate from the other notebook, could you define Hogg and Lang 2013 explicitly here?

Reply:
Done!